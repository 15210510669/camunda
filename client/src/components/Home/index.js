/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

export {default as Home} from './Home';
export {default as Collection} from './Collection';
