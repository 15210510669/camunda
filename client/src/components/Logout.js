/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {useEffect} from 'react';
import {withRouter} from 'react-router-dom';
import {get} from 'request';
import {withErrorHandling} from 'HOC';
import {addNotification} from 'notifications';
import {t} from 'translation';

export function Logout({mightFail, history}) {
  useEffect(() => {
    mightFail(
      get('api/authentication/logout'),
      () => {
        addNotification({text: t('navigation.logoutSuccess')});
        history.replace('/');
      },
      () => {
        addNotification({text: t('navigation.logoutFailed'), type: 'error'});
        history.replace('/');
      }
    );
  }, [mightFail, history]);

  return null;
}

export default withRouter(withErrorHandling(Logout));
