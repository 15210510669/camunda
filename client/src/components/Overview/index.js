export {default as Overview} from './Overview';
export {default as Alerts} from './Alerts';
