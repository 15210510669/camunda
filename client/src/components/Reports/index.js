export {default as Reports} from './Reports';
export {default as Report} from './Report';
export {Filter} from './filter'
