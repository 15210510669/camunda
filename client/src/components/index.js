export {Footer} from './Footer';
export {Header, LogoutButton} from './Header';
export {Home} from './Home';
export {Login} from './Login';
export {PrivateRoute} from './PrivateRoute';
export {Dashboard} from './Dashboard';
export {Dashboards} from './Dashboards';
export {Report} from './Report';
export {Reports} from './Reports';
