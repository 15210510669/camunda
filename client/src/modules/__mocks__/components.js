export {default as Button} from '../components/__mocks__/Button';
export {default as Input} from '../components/__mocks__/Input';
export {default as Modal} from '../components/__mocks__/Modal';
export {default as Popover} from '../components/__mocks__/Popover';
export {default as ControlGroup} from '../components/__mocks__/ControlGroup';
export {default as ShareEntity} from '../components/__mocks__/ShareEntity';
export {default as DashboardView} from '../components/__mocks__/DashboardView';
export {default as Icon} from '../components/__mocks__/Icon';
export {default as Dropdown} from '../components/__mocks__/Dropdown';
export {default as Message} from '../components/__mocks__/Message';
export {default as ReportView} from '../components/__mocks__/ReportView';
export {default as Select} from '../components/__mocks__/Select';
export {default as BPMNDiagram} from '../components/__mocks__/BPMNDiagram';
export {default as TargetValueBadge} from '../components/__mocks__/TargetValueBadge';
export {default as Table} from '../components/__mocks__/Table';
export {default as Logo} from '../components/__mocks__/Logo';
export {default as ErrorMessage} from '../components/__mocks__/ErrorMessage';
