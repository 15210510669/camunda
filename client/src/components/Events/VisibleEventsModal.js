/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import React, {useState} from 'react';

import {Form, LabeledInput, Modal, MessageBox, Button, DocsLink} from 'components';
import {t} from 'translation';

export default function VisibleEventsModal({initialScope, onClose, onConfirm}) {
  const [scope, setScope] = useState(initialScope);

  const toggleScopeItem = (item) => {
    if (scope.includes(item)) {
      return setScope(scope.filter((i) => i !== item));
    }
    return setScope([...scope, item]);
  };

  const updateSource = () => scope.length > 0 && onConfirm(scope);

  return (
    <Modal open onClose={onClose} onConfirm={updateSource}>
      <Modal.Header>{t('events.sources.editScope')}</Modal.Header>
      <Modal.Content>
        <Form description={t('events.sources.eventListTip')}>
          <Form.Group>
            <LabeledInput
              label={t('events.sources.startAndEnd')}
              checked={scope.includes('process_instance')}
              onChange={() => toggleScopeItem('process_instance')}
              type="checkbox"
            />
            <LabeledInput
              checked={scope.includes('start_end')}
              onChange={() => toggleScopeItem('start_end')}
              label={t('events.sources.flownodeEvents')}
              type="checkbox"
            />
            <LabeledInput
              label={t('events.sources.allEvents')}
              checked={scope.includes('all')}
              onChange={() => toggleScopeItem('all')}
              type="checkbox"
            />
          </Form.Group>
          <MessageBox type="warning">
            {t('events.sources.eventListChangeWarning')}{' '}
            <DocsLink location="components/optimize/userguide/additional-features/event-based-processes/#camunda-events">
              {t('events.sources.learnMore')}
            </DocsLink>
          </MessageBox>
        </Form>
      </Modal.Content>
      <Modal.Actions>
        <Button main className="close" onClick={onClose}>
          {t('common.cancel')}
        </Button>
        <Button disabled={!scope.length} main primary className="confirm" onClick={updateSource}>
          {t('common.update')}
        </Button>
      </Modal.Actions>
    </Modal>
  );
}
