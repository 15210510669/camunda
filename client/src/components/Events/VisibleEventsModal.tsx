/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {useState} from 'react';
import {Button} from '@carbon/react';

import {Form, LabeledInput, CarbonModal as Modal, MessageBox, DocsLink} from 'components';
import {t} from 'translation';

interface VisibleEventsModalProps {
  initialScope: string[];
  onClose: () => void;
  onConfirm: (scope: string[]) => void;
}

export default function VisibleEventsModal({
  initialScope,
  onClose,
  onConfirm,
}: VisibleEventsModalProps) {
  const [scope, setScope] = useState<string[]>(initialScope);

  const toggleScopeItem = (item: string) => {
    if (scope.includes(item)) {
      return setScope(scope.filter((i) => i !== item));
    }
    return setScope([...scope, item]);
  };

  const updateSource = () => scope.length > 0 && onConfirm(scope);

  return (
    <Modal open onClose={onClose}>
      <Modal.Header>{t('events.sources.editScope')}</Modal.Header>
      <Modal.Content>
        <Form description={t('events.sources.eventListTip')}>
          <Form.Group>
            <LabeledInput
              label={t('events.sources.startAndEnd')}
              checked={scope.includes('process_instance')}
              onChange={() => toggleScopeItem('process_instance')}
              type="checkbox"
            />
            <LabeledInput
              checked={scope.includes('start_end')}
              onChange={() => toggleScopeItem('start_end')}
              label={t('events.sources.flownodeEvents')}
              type="checkbox"
            />
            <LabeledInput
              label={t('events.sources.allEvents')}
              checked={scope.includes('all')}
              onChange={() => toggleScopeItem('all')}
              type="checkbox"
            />
          </Form.Group>
          <MessageBox type="warning">
            {t('events.sources.eventListChangeWarning')}{' '}
            <DocsLink location="components/userguide/additional-features/event-based-processes/#camunda-events">
              {t('events.sources.learnMore')}
            </DocsLink>
          </MessageBox>
        </Form>
      </Modal.Content>
      <Modal.Footer>
        <Button kind="secondary" className="close" onClick={onClose}>
          {t('common.cancel')}
        </Button>
        <Button disabled={!scope.length} className="confirm" onClick={updateSource}>
          {t('common.update')}
        </Button>
      </Modal.Footer>
    </Modal>
  );
}
