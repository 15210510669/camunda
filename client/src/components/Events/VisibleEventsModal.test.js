/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import React from 'react';
import {shallow} from 'enzyme';

import VisibleEventsModal from './VisibleEventsModal';
import {LabeledInput, Button} from 'components';

const props = {
  initialScope: ['process_instance'],
  onConfirm: jest.fn(),
};

beforeEach(() => {
  props.onConfirm.mockClear();
});

it('should match snapshot', () => {
  const node = shallow(<VisibleEventsModal {...props} />);

  expect(node.find('Form')).toMatchSnapshot();
});

it('should add a scope', () => {
  const node = shallow(<VisibleEventsModal {...props} />);

  node.find(LabeledInput).at(1).simulate('change');

  node.find(Button).at(1).simulate('click');

  expect(props.onConfirm).toHaveBeenCalledWith(['process_instance', 'start_end']);
});

it('should disable the confirm button when no scope is selected', () => {
  const node = shallow(<VisibleEventsModal {...props} />);

  node.find(LabeledInput).at(0).simulate('change');

  expect(node.find(Button).at(1)).toBeDisabled();
});
