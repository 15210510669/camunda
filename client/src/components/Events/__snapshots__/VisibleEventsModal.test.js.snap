// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Form
  description="Configure which events are displayed in the table."
>
  <FormGroup>
    <ForwardRef(LabeledInput)
      checked={true}
      label="Process start and end"
      onChange={[Function]}
      type="checkbox"
    />
    <ForwardRef(LabeledInput)
      checked={false}
      label="Start and End Flownode Events"
      onChange={[Function]}
      type="checkbox"
    />
    <ForwardRef(LabeledInput)
      checked={false}
      label="All Flownode Events"
      onChange={[Function]}
      type="checkbox"
    />
  </FormGroup>
  <MessageBox
    type="warning"
  >
    Changing the event list may add or remove events. Existing event mappings may be removed.
     
    <Component
      location="components/userguide/additional-features/event-based-processes/#camunda-events"
    >
      Learn more
    </Component>
  </MessageBox>
</Form>
`;
